youtube_dl/extractor/drtuber.py

```diff
@@ -44,8 +44,25 @@ def _real_extract(self, url):
         webpage = self._download_webpage(
             'http://www.drtuber.com/video/%s' % video_id, display_id)
 
-        video_url = self._html_search_regex(
-            r'<source src="([^"]+)"', webpage, 'video URL')
+        video_data = self._download_json('http://www.drtuber.com/player_config_json/', video_id, query={
+            'vid': video_id,
+            'embed': 0,
+            'aid': 0,
+            'domain_id': 0,
+        })
+
+        formats = []
+        for video_key, video_value in video_data.items():
+            if video_key == 'files':
+                for video_url_key, video_url_value in video_data[video_key].items():
+                    if video_url_value:
+                        formats.append({
+                            'format_id': video_url_key.upper(),
+                            'quality': 2 if video_url_key == 'hq' else 1,
+                            'url': video_url_value
+                        })
+        self._check_formats(formats, video_id)
```

Comment:
"Add test for a video with unavailable formats."

Can you please clarify?  Are you referring to a test case in which the video URL yields a 404 NotFound or otherwise generic 'not available' error?